File: packages/ts/shared-schema/src/entities/battle.ts
export type BattleEndConditionType = 'ALL_SLOTS_FILLED' | 'TURN_LIMIT';

export interface BattleEndCondition {
  type: BattleEndConditionType;
  turnLimit?: number;
}

export interface BattleDefinition {
  id: string;
  formatVersion: string;
  name: string;
  battlefieldConfigId: string;
  playerDeckConfigId: string;
  opponentDeckConfigId: string;
  rulesConfigId: string;
  endCondition: BattleEndCondition;
}

File: packages/py/forge-core/src/forge_core/domain/entities/battle.py
from typing import Literal

from .base import CamelModel as BaseModel


class BattleEndCondition(BaseModel):
    type: Literal["ALL_SLOTS_FILLED", "TURN_LIMIT"]
    turn_limit: int | None = None


class BattleDefinition(BaseModel):
    id: str
    format_version: str
    name: str
    battlefield_config_id: str
    player_deck_config_id: str
    opponent_deck_config_id: str
    rules_config_id: str
    end_condition: BattleEndCondition
